Creation of entity with relationship fails on latest beta

App used is same as https://github.com/jhipster/generator-jhipster/issues/14158

Try creating an entity with a relationship


I'll add in more details later, creating the issue to not loose track of it

@pascalgrimaud @avdev4j if this is something alreday fixed on main let me know. I'll test on main tomorrow and update the ticket otherwise. Too late today

So I just created a new app using `jhipster jdl e-commerce-monolith.jdl` and ran `./gradlew` and it fails on first try with a Typescript error. I'm gonna look into that first and proceed from there
```
ERROR in /home/deepu/workspace/temp/ingApp/src/main/webapp/app/entities/shopping-cart/shopping-cart-update.tsx
  58:47  error  'customerDetails' is already declared in the upper scope  @typescript-eslint/no-shadow

```

--- a/generators/client/index.js
+++ b/generators/client/index.js
@@ -345,7 +345,7 @@ module.exports = class JHipsterClientGenerator extends BaseBlueprintGenerator {
               'echo "Waiting for server at port $npm_package_config_backend_port to start" && wait-on http-get://localhost:$npm_package_config_backend_port/management/health && echo "Server at port $npm_package_config_backend_port started"',
             'pree2e:headless': 'npm run ci:server:await',
             'ci:e2e:run': 'concurrently -k -s first "npm run ci:e2e:server:start" "npm run e2e:headless"',
-            'e2e:dev': 'concurrently -k -s first "./mvnw" "e2e:run"',
+            'e2e:dev': 'concurrently -k -s first "./mvnw" "e2e"',
           });
         }
       },

--- a/generators/cypress/templates/src/test/javascript/cypress/support/entity.ts.ejs
+++ b/generators/cypress/templates/src/test/javascript/cypress/support/entity.ts.ejs
@@ -47,14 +47,14 @@ Cypress.Commands.add('setFieldImageAsBytesOfEntity', (fieldName: string, fileNam
       const list = new DataTransfer();
       list.items.add(file);
       const myFileList = list.files;
-      el[0].files = myFileList;
+      (el[0] as HTMLInputElement).files = myFileList;
       el[0].dispatchEvent(new Event('change', {bubbles: true}));
     });
 });
 
 Cypress.Commands.add('setFieldSelectToLastOfEntity', (fieldName: string) => {
   return cy.get(`[data-cy="${fieldName}"]`).then(select => {
-    const selectSize = Number(select.attr("size"));
+    const selectSize = (select[0] as HTMLSelectElement)?.options?.length || Number(select.attr('size')) || 0;
     if (selectSize > 0) {
       return cy.get(`[data-cy="${fieldName}"] option`).then((options: any) => {
         const elements = [...options].map(o => o.label);

--- a/generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs
+++ b/generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs
@@ -117,9 +117,6 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
   <%_ manyToManyOwners.forEach(val => { _%>
   const [ids<%= val.relationshipName %>, setIds<%= val.relationshipName %>] = useState([]);
   <%_ }) _%>
-  <%_ relFieldNames.forEach(val => { _%>
-  const [<%= val.relationshipFieldName %>Id, set<%= upperFirstCamelCase(val.relationshipFieldName) %>Id] = useState('0');
-  <%_ }) _%>
   const [isNew] = useState(!props.match.params || !props.match.params.id);
 
   const { <%= entityInstance %>Entity,<%
@@ -147,16 +144,6 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
   const handleClose = () => {
     props.history.push('/<%= entityFileName %>'<%_ if (pagination === 'pagination') { _%> + props.location.search<%_ } _%>);
   }
-
-  <%_ relFieldNames.forEach(val => { _%>
-    <%_ if (val.relationshipType === 'many-to-one')  { _%>
-  useEffect(() => {
-    if (<%= val.relationshipFieldName %>Id === '0' && !isNew) {
-      set<%= upperFirstCamelCase(val.relationshipName) %>Id(<%= entityInstance %>Entity?.<%= val.relationshipName %>?.id.toString());
-    }
-  }, [<%= entityInstance %>Entity]);
-    <%_ } _%>
-  <%_ }) _%>
 
   useEffect(() => {
     <%_ if (pagination === 'infinite-scroll') { _%>
@@ -205,21 +192,21 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
     if (errors.length === 0) {
       const entity = {
         ...<%= entityInstance %>Entity,
-        ...values
+        ...values,
         <%_ manyToManyOwners.forEach(rel => { _%>
-        , <%= rel.relationshipFieldNamePlural %>: mapIdList(values.<%= rel.relationshipFieldNamePlural %>)
+        <%= rel.relationshipFieldNamePlural %>: mapIdList(values.<%= rel.relationshipFieldNamePlural %>),
         <%_ }) _%>
         <%_ relFieldNames.forEach(rel => { _%>
           <%_ if (rel.relationshipType === 'many-to-one' && rel.otherEntityName !== 'user')  { _%>
-          , <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(it => it.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString())
+          <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(it => it.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString()),
           <%_ } _%>
         <%_ }) _%>
+        <%_ if (uniqueRelationFields.has('users')) { _%>
+          <%_ relationships.filter(rel => rel.relationshipType === 'one-to-one' && rel.ownerSide === true && rel.otherEntityName === 'user').forEach(rel => { _%>
+        <%= rel.relationshipFieldName %>: users.find(user => user.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString()),
+          <%_ }) _%>
+        <%_ } _%>
       }
-      <%_ if (uniqueRelationFields.has('users') && authenticationType === 'oauth2' && applicationType === 'gateway') { _%>
-        <%_ relationships.filter(rel => rel.relationshipType === 'one-to-one' && rel.ownerSide === true && rel.otherEntityName === 'user').forEach(rel => { _%>
-      entity.<%= rel.relationshipFieldName %> = users.find(user => user.id.toString() === values.<%= rel.relationshipFieldName %>.id.toString());
-        <%_ }) _%>
-      <%_ } _%>
 
       if (isNew) {
         props.createEntity(entity);
@@ -414,18 +401,20 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
               const otherEntityFieldCapitalized = rel.otherEntityFieldCapitalized;
               const relationshipRequired = rel.relationshipRequired;
               const translationKey = `${i18nKeyPrefix}.${relationshipName}`; _%>
-              <%_ if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true && otherEntityName === 'user')) { _%>
+              <%_ if (relationshipType === 'many-to-one' || (relationshipType === 'one-to-one' && ownerSide === true)) { _%>
           <AvGroup>
             <Label for="<%= entityFileName %>-<%= relationshipName %>">
               <Translate contentKey="<%= translationKey %>"><%= relationshipNameHumanized %></Translate>
             </Label>
-                <%_ if (!relationshipRequired) { _%>
             <AvInput
               id="<%= entityFileName %>-<%= relationshipName %>"
               data-cy="<%= relationshipFieldName %>"
               type="select"
               className="form-control"
-              name="<%= relationshipFieldName %>.id"
+              name="<%= relationshipFieldName %>Id"
+              <%_ if (relationshipRequired) { _%>
+              required
+              <%_ } _%>
             >
               <option value="" key="0" />
               {
@@ -438,73 +427,9 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
                 ) : null
               }
             </AvInput>
-            <%_ } else { _%>
-              <%_ if (!relationshipRequired) { _%>
-                <AvInput
-                  id="<%= entityFileName %>-<%= relationshipName %>"
-                  data-cy="<%= relationshipFieldName %>"
-                  type="select"
-                  className="form-control"
-                  name="<%= relationshipFieldName %>Id"
-                >
-                  <option value="" key="0" />
-                  {
-                    <%= otherEntityNamePlural %> ? <%= otherEntityNamePlural %>.map(otherEntity =>
-                      <option
-                        value={otherEntity.<%= otherEntityPkName %>}
-                        key={otherEntity.<%= otherEntityPkName %>}>
-                        {otherEntity.<%= otherEntityField %>}
-                      </option>
-                    ) : null
-                  }
-                </AvInput>
-              <%_ } else { _%>
-                <AvInput
-                  id="<%= entityFileName %>-<%= relationshipName %>"
-                  data-cy="<%= relationshipFieldName %>"
-                  type="select"
-                  className="form-control"
-                  name="<%= relationshipFieldName %>Id"
-                  value={<%= relationshipFieldName %>Id}
-                  required
-                >
-                  {
-                    <%= otherEntityNamePlural %> ? <%= otherEntityNamePlural %>.map(otherEntity =>
-                      <option
-                        value={otherEntity.<%= otherEntityPkName %>}
-                        key={otherEntity.<%= otherEntityPkName %>}>
-                        {otherEntity.<%= otherEntityField %>}
-                      </option>
-                    ) : null
-                  }
-                </AvInput>
-                <AvFeedback><Translate contentKey="entity.validation.required">This field is required.</Translate></AvFeedback>
-              <%_ } _%>
+            <%_ if (relationshipRequired) { _%>
+            <AvFeedback><Translate contentKey="entity.validation.required">This field is required.</Translate></AvFeedback>
             <%_ } _%>
-          </AvGroup>
-              <%_ } else if (relationshipType === 'one-to-one' && ownerSide === true) { _%>
-          <AvGroup>
-            <Label for="<%= entityFileName %>-<%= relationshipName %>">
-              <Translate contentKey="<%= translationKey %>"><%= relationshipNameHumanized %></Translate>
-            </Label>
-                <AvInput
-                  id="<%= entityFileName %>-<%= relationshipName %>"
-                  data-cy="<%= relationshipFieldName %>"
-                  type="select"
-                  className="form-control"
-                  name="<%= relationshipFieldName %>.id"
-                >
-                  <option value="" key="0" />
-                  {
-                    <%= otherEntityNamePlural %> ? <%= otherEntityNamePlural %>.map(otherEntity =>
-                      <option
-                        value={otherEntity.<%= otherEntityPkName %>}
-                        key={otherEntity.<%= otherEntityPkName %>}>
-                        {otherEntity.<%= otherEntityField %>}
-                      </option>
-                    ) : null
-                  }
-                </AvInput>
           </AvGroup>
               <%_ } else if (relationshipType === 'many-to-many' && rel.ownerSide === true) { _%>
           <AvGroup>

--- a/generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs
+++ b/generators/entity-client/templates/react/src/main/webapp/app/entities/entity-update.tsx.ejs
@@ -147,7 +147,7 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
   const handleClose = () => {
     props.history.push('/<%= entityFileName %>'<%_ if (pagination === 'pagination') { _%> + props.location.search<%_ } _%>);
   }
-    
+
   <%_ relFieldNames.forEach(val => { _%>
     <%_ if (val.relationshipType === 'many-to-one')  { _%>
   useEffect(() => {
@@ -211,7 +211,7 @@ export const <%= entityReactName %>Update = (props: I<%= entityReactName %>Updat
         <%_ }) _%>
         <%_ relFieldNames.forEach(rel => { _%>
           <%_ if (rel.relationshipType === 'many-to-one' && rel.otherEntityName !== 'user')  { _%>
-          , <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(<%= rel.relationshipFieldName %> => <%= rel.relationshipFieldName %>.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString())
+          , <%= rel.relationshipFieldName %>: <%= rel.otherEntityNamePlural %>.find(it => it.id.toString() === values.<%= rel.relationshipFieldName %>Id.toString())
           <%_ } _%>
         <%_ }) _%>
       }

